src/imagery/i.eodag/i.eodag.py

```diff
@@ -215,54 +298,142 @@ def no_fallback_search(search_parameters, provider):
     # TODO: Would it be useful if user could iterate through
     # the pages manually, and look for the product themselves?
     try:
-        return list(search_result)[0]
+        # Merging the pages into one list with all products
+        return [j for i in search_result for j in i]
     except Exception as e:
         gs.verbose(e)
-        gs.fatal(_("Server error, try again."))
+        gs.fatal(_("Server error, please try again."))
 
 
-def main():
-    # products: https://github.com/CS-SI/eodag/blob/develop/eodag/resources/product_types.yml
+def create_products_dataframe(eo_products):
+    result_dict = {"id": [], "time": [], "cloud_coverage": [], "product_type": []}
+    for product in eo_products:
+        if "id" in product.properties and product.properties["id"] is not None:
+            result_dict["id"].append(product.properties["id"])
+        else:
+            result_dict["id"].append(None)
+        if (
+            "startTimeFromAscendingNode" in product.properties
+            and product.properties["startTimeFromAscendingNode"] is not None
+        ):
+            try:
+                result_dict["time"].append(
+                    normalize_time(product.properties["startTimeFromAscendingNode"])
+                )
+            except:
+                result_dict["time"].append(
+                    product.properties["startTimeFromAscendingNode"]
+                )
+        else:
+            result_dict["time"].append(None)
+        if (
+            "cloudCover" in product.properties
+            and product.properties["cloudCover"] is not None
+        ):
+            result_dict["cloud_coverage"].append(product.properties["cloudCover"])
+        else:
+            result_dict["cloud_coverage"].append(None)
+        if (
+            "productType" in product.properties
+            and product.properties["productType"] is not None
+        ):
+            result_dict["product_type"].append(product.properties["productType"])
+        else:
+            result_dict["product_type"].append(None)
```

Comment:
This could be simplified along these lines:
```suggestion
    result_dict = {"id": [], "time": [], "cloudCover": [], "productType": []}
    for product in eo_products:
        for key in result_dict:
            if key == "time":
                try:
                    result_dict[key].append(normalize_time(product.properties["startTimeFromAscendingNode"]))
                except:
                    result_dict["time"].append(product.properties.get("startTimeFromAscendingNode"))

            else:
                result_dict[key].append(product.properties.get(key)
```

Please not the change in the dict keys and the need to change them in the rest of the code if you change them here....

That said, I am still not sure if a data frame is needed given sthe sort and filter ("crunch") capabilities in EODAG....